### ISSUE TITLE
linux-5.4.88-modules (aarch64) fails, blocking nixos-unstable

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#109252]
**Describe the bug**
https://hydra.nixos.org/build/135054936

Caused by https://github.com/NixOS/nixpkgs/pull/78430.

**To Reproduce**
1. ...
2. ...
3. ...

**Expected behavior**

**Screenshots**

**Additional context**

**Notify maintainers**

cc @NeQuissimus 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by primeos** [comment]
[Full log](https://nix-cache.s3.amazonaws.com/log/1yhfy63iaxg1adlzwp91bxf6076863ij-linux-5.4.88-modules-shrunk.drv)

The error:
```
root module: vmw_balloon
modprobe: FATAL: Module vmw_balloon not found in directory /nix/store/877vj59c34mrwa7b6120hdf02rkb5y08-linux-5.4.88-modules/lib/modules/5.4.88
builder for '/nix/store/1yhfy63iaxg1adlzwp91bxf6076863ij-linux-5.4.88-modules-shrunk.drv' failed with exit code 1
```

Seems to be related to `nixos/modules/profiles/all-hardware.nix` (`vmw_balloon` was added in 870375e19d1ec3ad69de7eb1c73c6c80cc214108):
```nix
      # VMware support.
      "mptspi" "vmw_balloon" "vmwgfx" "vmw_vmci" "vmw_vsock_vmci_transport" "vmxnet3" "vsock"
```

Not sure why the `vmw_balloon` kernel module seems to be suddenly gone.

cc @volth (disabled by GitHub...)

**Comment by primeos** [comment]
**Update:** Ok, so I got a bit confused by the Hydra UI as e.g. https://hydra.nixos.org/build/135057055 states "System: x86_64-linux" but the failing build is from an `aarch64-linux` machine (`e2da2215`). I guess this is because we don't have a dedicated channel for AArch64.

In any case as long as the failure only occurs on AArch64 builds that would explain it as the `vmw_balloon` module depends on x86):
```
config VMWARE_BALLOON
        tristate "VMware Balloon Driver"
        depends on VMWARE_VMCI && X86 && HYPERVISOR_GUEST
        select MEMORY_BALLOON
        help
          This is VMware physical memory management driver which acts
          like a "balloon" that can be inflated to reclaim physical pages
          by reserving them in the guest and invalidating them in the
          monitor, freeing up the underlying machine pages so they can
          be allocated to other guests. The balloon can also be deflated
          to allow the guest to use more physical memory.

          If unsure, say N.

          To compile this driver as a module, choose M here: the
          module will be called vmw_balloon.
```

But the questions still is what changed in Nixpkgs to cause this regression.

**Comment by FRidh** [comment]
maybe just the kernel update? a9dfa241f4380a9a9164687d55a56a424ae8537a

**Comment by primeos** [comment]
Think I finally found it: fdf32154fc90698ef72ffa96ef39d920e1b9c951

cc @puckipedia

**Comment by primeos** [comment]
So I guess it would be best to revert that commit in master for now and try to reland it via staging/staging-next as [log](https://nix-cache.s3.amazonaws.com/log/09w1wqcclc75sfnfrj9xg68nrki26pyx-linux-5.4.88-modules-shrunk.drv) contains multiple modules that are "not found" (10 matches) and this might also affect other things. But we could of course also try to fix the errors in master instead if someone prefers that.

**Comment by vcunat** [comment]
I already have code that adds all those conditionals to make the minimal ISO build, so we could use that if that's the only issue.

**Comment by FRidh** [comment]
> I already have code that adds all those conditionals to make the minimal ISO build, so we could use that if that's the only issue.

suggest we try and see

**Comment by vcunat** [comment]
> Ok, so I got a bit confused by the Hydra UI as e.g. https://hydra.nixos.org/build/135057055 states "System: x86_64-linux" but the failing build is from an `aarch64-linux` machine (`e2da2215`). I guess this is because we don't have a dedicated channel for AArch64.

You're looking at the `tested` meta-job that combines things from multiple platforms.

**Comment by primeos** [comment]
> You're looking at the `tested` meta-job that combines things from multiple platforms.

Yeah, I didn't know that it combines things from multiple platforms this way (thought there would be two `tested` jobsets) but I guess that makes sense in hindsight. But in any case it would IMO be nice to have dedicated channels for each architecture (cc #83049) but that is another topic (and I didn't really look a the details).

Anyway, thanks for pushing a fix :)

**Comment by vcunat** [comment]
:-) The `trunk-combined` jobset bears lots of history, just look at the name from svn times...
